### PR TITLE
Limit ucr columns to 63 characters

### PR DESCRIPTION
**Pull request by TylerSheffels**
@orangejenny 

refer to: http://manage.dimagi.com/default.asp?168908

### CODE REVIEW

File: corehq/apps/userreports/tests/test_utils.py [hunk 1]
```diff
@@ -17,7 +17,11 @@ def test_column_trickery(self):
...
         trap2 = get_column_name("its_a/trap")
         trap3 = get_column_name("its_a_trap")
+        trap4 = get_column_name("its/a/path/that/also/happens/to/be/a/bunch/longer/than/sixty/three/characters")
+        trap4_expected = 'to_be_a_bunch_longer_than_sixty_three_characters_f16c5226'
 
         self.assertNotEqual(real, trap1)
         self.assertNotEqual(real, trap2)
         self.assertNotEqual(real, trap3)
+        self.assertEqual(trap4, trap4_expected)
+
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_utils.py]: W391 blank line at end of file

File: corehq/apps/userreports/sql.py [hunk 2]
```diff
@@ -114,7 +123,17 @@ def _hash(parts):
         end = parts[-1]
         return hashlib.sha1('{}_{}'.format(hashlib.sha1(front).hexdigest(), end)).hexdigest()[:8]
 
-    return "_".join(parts + [_hash(parts)])
+    column_len = 0
+    new_parts = []
+    parts.reverse()
+    for part in parts:
```
Comment (orangejenny) [inline corehq/apps/userreports/sql.py]: Could this loop be dropped by doing something like `path = path[-54:]` before the `parts = path.split("/")` above?

Comment (TylerSheffels) [inline corehq/apps/userreports/sql.py]: First, I didn't want to affect the generated hash.  Second I wanted to only use whole parts of the path, but I agree this is not super pretty.

**Comment by orangejenny** [comment]
fine once tests pass

**Comment by czue** [comment]
build failure doesn't look related - kicked

File: corehq/apps/userreports/tests/test_utils.py [hunk 3]
```diff
@@ -17,7 +17,10 @@ def test_column_trickery(self):
         trap1 = get_column_name("its/a_trap")
         trap2 = get_column_name("its_a/trap")
         trap3 = get_column_name("its_a_trap")
+        trap4 = get_column_name("its/a/path/that/also/happens/to/be/a/bunch/longer/than/sixty/three/characters")
+        trap4_expected = 'ppens_to_be_a_bunch_longer_than_sixty_three_characters_6174b354'
```
Comment (czue) [inline corehq/apps/userreports/tests/test_utils.py]: :+1:

**Comment by czue** [comment]
(ran ucr tests locally) :white_check_mark:

**Pull request merged**